### ISSUE TITLE
Last gene2go update failed

### ISSUE THREAD

**Issue by arteymix** [PavlidisLab/rdp#397]
```
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: 2023-11-01 00:02:53.892 ERROR 29481 --- [   scheduling-1] ubc.pavlab.rdp.services.GOServiceImpl    : Failed to parse gene2go annotations from URL [ftp://ftp.ncbi.nlm.nih.gov/gene/DATA/gene2go.gz
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: ubc.pavlab.rdp.util.ParseException: 10,799,745: Unexpected number of parts in: 88036        112347737        GO:0047560        IEA        88036        1123ion
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at ubc.pavlab.rdp.util.UncheckedParseException.<init>(UncheckedParseException.java:16) ~[classes!/:1.5.7]
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at ubc.pavlab.rdp.util.Gene2GoParser$Record.parseLine(Gene2GoParser.java:37) ~[classes!/:1.5.7]
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at ubc.pavlab.rdp.util.Gene2GoParser.lambda$parse$0(Gene2GoParser.java:67) ~[classes!/:1.5.7]
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193) ~[na:1.8.0_382]
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at java.util.Iterator.forEachRemaining(Iterator.java:116) ~[na:1.8.0_382]
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801) ~[na:1.8.0_382]
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482) ~[na:1.8.0_382]
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472) ~[na:1.8.0_382]
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708) ~[na:1.8.0_382]
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[na:1.8.0_382]
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:566) ~[na:1.8.0_382]
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at ubc.pavlab.rdp.util.Gene2GoParser.parse(Gene2GoParser.java:68) ~[classes!/:1.5.7]
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at ubc.pavlab.rdp.services.GOServiceImpl.updateGoTerms(GOServiceImpl.java:162) ~[classes!/:1.5.7]
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at ubc.pavlab.rdp.services.CacheService.updateCache(CacheService.java:63) [classes!/:1.5.7]
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_382]
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_382]
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_382]
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_382]
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at org.springframework.scheduling.support.ScheduledMethodRunnable.run(ScheduledMethodRunnable.java:84) [spring-context-5.3.27.jar!/:5.3.27]
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54) [spring-context-5.3.27.jar!/:5.3.27]
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at org.springframework.scheduling.concurrent.ReschedulingRunnable.run(ReschedulingRunnable.java:95) [spring-context-5.3.27.jar!/:5.3.27]
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [na:1.8.0_382]
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at java.util.concurrent.FutureTask.run(FutureTask.java:266) [na:1.8.0_382]
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [na:1.8.0_382]
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [na:1.8.0_382]
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [na:1.8.0_382]
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [na:1.8.0_382]
Nov 01 00:02:53 moe.msl.ubc.ca java[29481]: at java.lang.Thread.run(Thread.java:750) [na:1.8.0_382]
```

This might be related to #394

**Comment by arteymix** [comment]
Fixed in e28bdd076f26be898b722f17f909f8071812d677

I still have to create a hotfix branch and backport it.

**Comment by arteymix** [comment]
Ok, it's in acdc2c930ebf6d78b42f5799c1be73b5e01e1a51.
